Traceback (most recent call last):
  File "/home/natalie/.local/lib/python3.8/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/home/natalie/.local/lib/python3.8/site-packages/nbclient/client.py", line 1204, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/home/natalie/.local/lib/python3.8/site-packages/nbclient/util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/home/natalie/.local/lib/python3.8/site-packages/nbclient/util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "/usr/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/home/natalie/.local/lib/python3.8/site-packages/nbclient/client.py", line 663, in async_execute
    await self.async_execute_cell(
  File "/home/natalie/.local/lib/python3.8/site-packages/nbclient/client.py", line 965, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/home/natalie/.local/lib/python3.8/site-packages/nbclient/client.py", line 862, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
data = {
    'cols': ['Kriegswaffen (St√ºck)', 'Lang- und Faustfeuerwaffen', 'Andere Waffen', 'Munition', 'Explosivstoffe (kg)'],
    2021: [40, 5.236, 6.198, 137.499, 36],
    2020: [34, 372, 448, 31.533, 1],
    2019: [30, 156, 202, 40.601, 8]
}


if len(data[2020]) != len(data[2019]) or len(data[2020]) == len(data[2021]):
    print('Fehler: √úberpr√ºfe die Dateneingabe: Die Anzahl an Werten pro Jahr stimmt nicht √ºberein.\n\n')

values_diff = []
for idx in range(len(data[2020])):
    diff = differenz(data[2020][idx], data[2019][idx])
    values_diff.append(diff)
    
if 'cols' in data:
    print('Prozentuale Differenz: \n')
    idx = 0
    for diff in values_diff:
        category = data['cols'][idx]
        print(f'{category: <28}: {diff} %')
    
    
def differenz(val1=100, val2=0):
    round(((val1 - val2) / val2 ) * 100, 1)
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mNameError[0m                                 Traceback (most recent call last)
Input [0;32mIn [6][0m, in [0;36m<cell line: 13>[0;34m()[0m
[1;32m     12[0m values_diff [38;5;241m=[39m []
[1;32m     13[0m [38;5;28;01mfor[39;00m idx [38;5;129;01min[39;00m [38;5;28mrange[39m([38;5;28mlen[39m(data[[38;5;241m2020[39m])):
[0;32m---> 14[0m     diff [38;5;241m=[39m [43mdifferenz[49m(data[[38;5;241m2020[39m][idx], data[[38;5;241m2019[39m][idx])
[1;32m     15[0m     values_diff[38;5;241m.[39mappend(diff)
[1;32m     17[0m [38;5;28;01mif[39;00m [38;5;124m'[39m[38;5;124mcols[39m[38;5;124m'[39m [38;5;129;01min[39;00m data:

[0;31mNameError[0m: name 'differenz' is not defined
NameError: name 'differenz' is not defined

